frontend/wasm/src/document.rs

```diff
@@ -92,16 +92,17 @@ pub fn close_all_documents_with_confirmation() -> Result<(), JsValue> {
 // TODO: Call event when the panels are resized
 /// Viewport resized
 #[wasm_bindgen]
-pub fn viewport_resize(new_width: u32, new_height: u32) -> Result<(), JsValue> {
+pub fn viewport_resize(new_width: f64, new_height: f64) -> Result<(), JsValue> {
 	let ev = InputPreprocessorMessage::ViewportResize((new_width, new_height).into());
 	EDITOR_STATE.with(|editor| editor.borrow_mut().handle_message(ev)).map_err(convert_error)
 }
 
 // TODO: When a mouse button is down that started in the viewport, this should trigger even when the mouse is outside the viewport (or even the browser window if the browser supports it)
 /// Mouse movement within the screenspace bounds of the viewport
 #[wasm_bindgen]
-pub fn on_mouse_move(x: u32, y: u32, modifiers: u8) -> Result<(), JsValue> {
+pub fn on_mouse_move(x: f64, y: f64, modifiers: u8) -> Result<(), JsValue> {
```

Comment:
I am not really a fan of doing the conversion here, I would prefer it to take place explicitly in the ipp.
(We might want to apply a different mapping based on the input device)

Reply:
The conversion from int to float? JS gives this as a float, not an int.

Reply:
Yeah it just hurts to specify pixels as floats but I guess there is nothing I can do about it

Reply:
Retina/display scaling, and perhaps pen inputs that have a different resolution digitizer than the screen resolution, are cases that come to mind. Not sure how the OS or browser treats it, but the JS spec clearly has a reason for specifying floats.